### PR TITLE
ci: Move merge GHA workflow to its own yaml file. WIP

### PR DESCRIPTION
**Pull request by novakzaballa**
Thanks for submitting a PR! Please check the boxes below:

- [X] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [X] I have added information to `docs/` if required so people know about the feature!
- [X] I have filled in the "Changes" section below?
- [X] I have filled in the "How did you test this code" section below?
- [X] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Moved the merge GHA workflow to its own yaml file so we can track the execution more easily. Also,  removed the corresponding code from the existing `api-pull-request.yml` file.

To make the workflow run we needed to ask @matthewelwell to change the repository settings in GitHub by adding `requested checks` as per the image below:
image

## How did you test this code?

Manually tested by merging a re-bumped version of this PR https://github.com/Flagsmith/flagsmith/pull/3264

### CODE REVIEW

File: .github/workflows/api-merge.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  merge_group:
+    types: [checks_requested]
+
+defaults:
+  run:
+    working-directory: api
+
+jobs:
+  test:
+    runs-on: General-Purpose-8c-Runner
```
Comment (dabeeeenster) [inline .github/workflows/api-merge.yml:13]: What's this? Why is it different to our other runners?

File: .github/workflows/api-merge.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    name: API Unit Tests
+
+    services:
+      postgres:
+        image: postgres:11.12-alpine
```
Comment (dabeeeenster) [inline .github/workflows/api-merge.yml:18]: this needs bumping as per https://github.com/Flagsmith/flagsmith/pull/3280

File: .github/workflows/api-merge.yml (hunk 1 above)
Comment (novakzaballa) [reply .github/workflows/api-merge.yml:13]: I'm testing a new runner setup by Matt for this purpose

File: .github/workflows/api-merge.yml (hunk 2 above)
Comment (novakzaballa) [reply .github/workflows/api-merge.yml:18]: Good catch, thank you.

File: .github/workflows/api-merge.yml (hunk 1 above)
Comment (novakzaballa) [reply .github/workflows/api-merge.yml:13]: The goal of this PR is meanly to test and monitor how `testmon` and `merge-queue` are working together.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**